railties/lib/rails/generators/rails/app/templates/config/initializers/hsts_subdomains.rb

```diff
@@ -0,0 +1 @@
+Rails.application.config.ssl_options.reverse_merge!({ hsts: { subdomains: true } })
```

Comment:
We can just set the ssl_option directly, not need to merge. This line will be generated only in new applications, so there is no `ssl_options` set anywhere. For existent applications we will let the deprecation warning handle it.

We will need to update the `config_when_updating` method in the generators to not generate this file.

Also it should be named `ssl_options.rb`